### ISSUE TITLE
Code Debugging

### ISSUE THREAD

**Issue by zengzhi2015** [JeniaNovellusDx/deeplabv3_keras#17]
Thanks for your great efforts so that I can use your code to build a new structure upon DeepLabv3p easily. But I do find a few bugs inside your code. I found them by debugging line by line. 
In my own project, I corrected these bugs but also reshaped a bit the structure of the code, like defining more functions for readability and modifying certain functions. 
However, I am not sure if it is proper to commit my modifications, since the modifications may violate the interface of your own project.
Would you like to give me a little advice? Thanks! By the way, I will finish the whole debugging in several weeks.

**Comment by ankits-curefit** [comment]
Hi @zengzhi2015  

I myself have been struggling to correct all the bugs in the code. I'm still stuck with 
`Index out of range using input dim 3; input has only 3 dims for 'loss/pred_mask_loss/strided_slice_1'` Not sure what's the issue.

Would you be able to share your working version?

**Comment by ankitksharma** [comment]
Hi @zengzhi2015 

I myself have been struggling to correct all the bugs in the code. I'm still stuck with
`Index out of range using input dim 3; input has only 3 dims for 'loss/pred_mask_loss/strided_slice_1' `
Not sure what's the issue.

Would you be able to share your working version?

**Comment by curryJ** [comment]
Could you tell me your keras version and tf version? There are some bugs when I use tf 2.0 and keras 2.2.4.I want to solve this Incompatible problem.Thank a lot!

**Comment by ckolluru** [comment]
I've been able to compile the model with tensorflow v1.11.0 and keras v2.2.4 

Using tensorflow v1.14.0 failed, creating the model resulted in a TypeError. Maybe try keras v2.3.1 when using tf v2.0? There are some compatibility issues:

https://stackoverflow.com/questions/62690377/tensorflow-compatibility-with-keras
